### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Fluder-Paradyne**

### CODE REVIEW

File: superagi/controllers/resources.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+from superagi.config.config import get_config
+from superagi.models.project import Project
+from starlette.responses import FileResponse
+from pathlib import Path
+from fastapi.responses import StreamingResponse
+
+router = APIRouter()
+
+
+# @router.post("/add/", status_code=201)
```
Comment (TransformerOptimus) [inline superagi/controllers/resources.py:21]: Remove these comments

File: superagi/agent/super_agi.py [hunk 2]
```diff
@@ -171,8 +194,22 @@ def execute(self, goals: List[str]):
...
         if action.name in tools:
             tool = tools[action.name]
+            print("_________________TESTING__________________")
+            print(action.name)
+            print(action.args)
             try:
                 observation = tool.execute(action.args)
+                print("Tool Observation : ")
+                print(observation)
+                if action.name == WRITE_FILE:
```
Comment (TransformerOptimus) [inline superagi/agent/super_agi.py:204]: Why is the tool code here. Add this inside tool.

File: superagi/agent/super_agi.py [hunk 3]
```diff
@@ -171,8 +194,22 @@ def execute(self, goals: List[str]):
             return "COMPLETE"
         if action.name in tools:
             tool = tools[action.name]
+            print("_________________TESTING__________________")
```
Comment (TransformerOptimus) [inline superagi/agent/super_agi.py:197]: remove these prints

File: superagi/controllers/agent_execution.py [hunk 4]
```diff
@@ -48,6 +51,8 @@ def get_agent_execution(agent_execution_id: int, Authorize: AuthJWT = Depends())
 @router.put("/update/{agent_execution_id}", response_model=sqlalchemy_to_pydantic(AgentExecution))
 def update_agent_execution(agent_execution_id: int,
                            agent_execution: sqlalchemy_to_pydantic(AgentExecution, exclude=["id"])):
+    print(agent_execution_id)
```
Comment (TransformerOptimus) [inline superagi/controllers/agent_execution.py:54]: remove these prints

File: alembic/versions/115a710c4685_init_db.py [hunk 5]
```diff
@@ -95,6 +95,7 @@ def upgrade() -> None:
         sa.Column('agent_execution_id', sa.Integer()),
```
Comment (TransformerOptimus) [inline alembic/versions/115a710c4685_init_db.py:95]: Remove the alembic folder.

Comment (luciferlinx101) [reply alembic/versions/115a710c4685_init_db.py:95]: removed

File: superagi/jobs/agent_executor.py [hunk 6]
```diff
@@ -55,49 +56,75 @@ def create_object(class_name, folder_name, file_name):
...
+            session = Session()
+            agent_execution = session.query(AgentExecution).filter(AgentExecution.id == agent_execution_id).first()
+            agent = session.query(Agent).filter(Agent.id == agent_execution.agent_id).first()
+            if agent_execution.status == "PAUSED" or agent_execution.status == "TERMINATED" \
+                    or agent_execution == "COMPLETED":
+                return
+
+            if not agent:
+                return "Agent Not found"
+            print("Agent Under Execution : ")
```
[the remark this answers is not in the document]
Comment (luciferlinx101) [reply superagi/jobs/agent_executor.py:71]: debug print statement removed

File: superagi/controllers/resources.py (hunk 1 above)
Comment (luciferlinx101) [reply superagi/controllers/resources.py:21]: Comments removed.

File: superagi/controllers/agent_execution.py (hunk 4 above)
Comment (luciferlinx101) [reply superagi/controllers/agent_execution.py:54]: removed

File: superagi/agent/super_agi.py (hunk 2 above)
Comment (luciferlinx101) [reply superagi/agent/super_agi.py:204]: after writing file to store in db it requires project id which is available here but if we shift this code to write tool under tools, project id would be a param and params of tool are generated by agent so project id can't be there so what to do?

File: superagi/controllers/resources.py [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+    file_name = resource.name
+
+    if not resource:
+        raise HTTPException(status_code=400, detail="Resource Not found!")
+
+    abs_file_path = Path(download_file_path).resolve()
+    if not abs_file_path.is_file():
+        raise HTTPException(status_code=404, detail="File not found")
+
+    print("Resource: ", resource_id)
```
[the remark this answers is not in the document]
Comment (luciferlinx101) [reply superagi/controllers/resources.py:122]: Removed

File: main.py [hunk 8]
```diff
@@ -218,30 +226,104 @@ def process_files(folder_path):
...
+    if response.ok:
+        data = response.json()
+        access_token = data.get('access_token')
+        print("Access Token : ", access_token)
+        github_api_url = 'https://api.github.com/user'
+        headers = {
+            'Authorization': f'Bearer {access_token}'
+        }
+        response = requests.get(github_api_url, headers=headers)
+        print("Response")
```
Comment (TransformerOptimus) [inline main.py:276]: remove prints

File: main.py [hunk 9]
```diff
@@ -218,30 +226,104 @@ def process_files(folder_path):
...
+#     return access_token
+
+@app.get('/github-login')
+def github_login():
+    github_client_id = ""
+    return RedirectResponse(f'https://github.com/login/oauth/authorize?scope=user:email&client_id={github_client_id}')
+
+@app.get('/github-auth')
+def github_auth_handler(code: str = Query(...),Authorize: AuthJWT = Depends()):
+    print(code)
```
Comment (TransformerOptimus) [inline main.py:253]: remove this

File: main.py [hunk 10]
```diff
@@ -218,30 +226,104 @@ def process_files(folder_path):
...
+        github_api_url = 'https://api.github.com/user'
+        headers = {
+            'Authorization': f'Bearer {access_token}'
+        }
+        response = requests.get(github_api_url, headers=headers)
+        print("Response")
+        print(response)
+        if response.ok:
+            user_data = response.json()
+            print("USER :",user_data)
```
Comment (TransformerOptimus) [inline main.py:280]: remove print

File: main.py [hunk 11]
```diff
@@ -218,30 +226,104 @@ def process_files(folder_path):
...
+#     access_token = Authorize.create_access_token(subject=user_email)
+#     return access_token
+
+@app.get('/github-login')
+def github_login():
+    github_client_id = ""
+    return RedirectResponse(f'https://github.com/login/oauth/authorize?scope=user:email&client_id={github_client_id}')
+
+@app.get('/github-auth')
+def github_auth_handler(code: str = Query(...),Authorize: AuthJWT = Depends()):
```
Comment (TransformerOptimus) [inline main.py:252]: Can you move this logic also to controllers.

**Review by TransformerOptimus — APPROVED** [review]

**Pull request merged**